src/rewriting.jl

```diff
@@ -20,26 +21,39 @@ RewritingSystem written as a list of pairs of `Word`s together with the ordering
 struct RewritingSystem{W<:AbstractWord, O<:Ordering} <: AbstractRewritingSystem{W, O}
     rwrules::Vector{Pair{W,W}}
     order::O
+    act::BitArray{1}
 end
 
+RewritingSystem(rwrules::Vector{Pair{W,W}}, order::O) where{W<:AbstractWord, O<:Ordering} = RewritingSystem(rwrules, order, trues(length(rwrules)))
+
 rules(s::RewritingSystem) = s.rwrules
+active(s::RewritingSystem) = s.act
+arules(s::RewritingSystem) = rules(s)[active(s)]
 ordering(s::RewritingSystem) = s.order
 
+isactive(s::RewritingSystem, i::Integer) = active(s)[i]
+
+setactive!(s::RewritingSystem, i::Integer) = active(s)[i] = true
+setinactive!(s::RewritingSystem, i::Integer) = active(s)[i] = false
+
 Base.:(==)(s::RewritingSystem, r::RewritingSystem) = (Set(rules(s)) == Set(rules(r)) && ordering(s) == ordering(r))
 Base.hash(s::RewritingSystem, h::UInt) =
     foldl((h, x) -> hash(x, h), s.rwrules, init = hash(s.order, hash(0x905098c1dcf219bc, h)))
 # the init value is simply hash(RewritingSystem)
 
-Base.push!(s::RewritingSystem{W,O}, r::Pair{W,W}) where {W<:AbstractWord, O<:Ordering} = (push!(rules(s), r); s)
-Base.pushfirst!(s::RewritingSystem{W,O}, r::Pair{W,W}) where {W<:AbstractWord, O<:Ordering} = (pushfirst!(rules(s), r); s)
+Base.push!(s::RewritingSystem{W,O}, r::Pair{W,W}) where {W<:AbstractWord, O<:Ordering} = (push!(rules(s), r); push!(active(s), true); s)
+Base.pushfirst!(s::RewritingSystem{W,O}, r::Pair{W,W}) where {W<:AbstractWord, O<:Ordering} = (pushfirst!(rules(s), r); pushfirst!(active(s), true); s)
+
+Base.pop!(s::RewritingSystem) = (deleteat!(active(s), length(s)); pop!(rules(s)))
```

Comment:
why not `Base.pop!(s::RewritingSystem) = (pop!(active(s)); pop!(rules(s)))`?

Reply:
Good question. I have no idea what craziness took hold of me. Corrected.